docs/preview/features/message-pumps/service-bus.md

```diff
@@ -302,6 +303,58 @@ string transactionId = correlationInfo.TransactionId;
 string operationId = correlationInfo.OperationId;
 ```
 
+## Automatic Azure Key Vault credentials rotation
+
+The additional library `Arcus.Messaging.Pumps.ServiceBus.KeyRotation` provides an extension on the message pump to restart the pump automatically when the credentials of the pump stored in Azure Key Vault are changed.
+This feature allows more reliable restarting instead of relying on authentication exceptions that may be throwed during the lifetime of the message pump.
+
+## How does this work?
+
+A background job will be run in the background which is in fact a message pump on a Service Bus Topic.
+The Azure Key Vault where the connection string for your target message pump is located should send `SecretNewVersionCreated` events to this Service Bus Topic.
+
+That way, when the background job receives a new Key Vault event, it will know that your target message pump should be restarted.
+
+### Installation
+
+This features requires to install our NuGet package:
+
+```shell
+PM > Install-Package Arcus.Messaging.Pumps.ServiceBus.KeyRotation
+```
+
+### Usage
+
+When the package is installed, you'll be able to use the extension in your application:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    // You should have a unique Job ID to identity the message pump so the automatic process knows which pump to restart.
+    string jobId = Guid.NewGuid().ToString();
+
+    string secretName = hostContext.Configuration["ARCUS_KEYVAULT_CONNECTIONSTRINGSECRETNAME"];
+    services.AddServiceBusQueueMessagePump(secretName, options => options.JobId = jobId)
+            .WithServiceBusMessageHandler<OrdersAzureServiceBusMessageHandler, Order>();
+
+    // This extension will be available to you once you installed the package.
+    services.WithAutoRestartOnRotatedCredentials(
```

Comment:
I would use this:
```suggestion
    services.WithAutoRestartServiceBusQueueMessagePumpOnRotatedCredentials(
```

Otherwise, it's not clear what you are autorestarting since it's not linked to https://github.com/arcus-azure/arcus.messaging/pull/107/files#diff-1f44b85c1f99fedb43b7e3a702a84b81d8ca6b38533d1a607ecec94295e9cc09R337

Reply:
We'll reduce it again as part of https://github.com/arcus-azure/arcus.messaging/issues/152

Reply:
If we'll reduce it, should we change it now?

Reply:
Oh, I see now, we're gonna make this one obsolete when we introduce a new one for the fluent type. Got it.